src/main/java/com/sobytylnik/ProfileController.java

```diff
@@ -35,9 +38,14 @@ public Profile createProfile(@RequestBody Profile newProfile) {
 
     @RequestMapping(value="/{id}", method = RequestMethod.PUT)
     public Optional<Profile> updateProfile(@PathVariable("id") Long id, @RequestBody Profile newProfile) {
-        newProfile.setId(id);
-        repository.merge(newProfile);
-        return repository.findById(id);
+        if(repository.existsById(id)){
+            newProfile.setId(id);
+            repository.save(newProfile);
+            return repository.findById(id);
```

Comment:
You can combine 2 lines in one:
`return Optional.of(repository.save(newProfile));`